tests/rptest/tests/cluster_config_test.py

```diff
@@ -1944,8 +1944,43 @@ def test_removal_of_legacy_default_defaulted(self, wipe_cache: bool):
         self._check_value_everywhere("retention_local_strict", True)
 
         # in >=23.3 (using upstream build) space management should be enabled by
-        # default provided that it wasn't explicitly disabled in 23.2. in this
-        # case no configs were changed so it should be enabled now.
+        # default provided that it wasn't explicitly disabled in 23.2.
+        # Strictness should remain enabled by default.
+        self._upgrade(wipe_cache)
+        self._check_value_everywhere("space_management_enable", True)
+        self._check_value_everywhere("retention_local_strict", True)
+
+        # survives a restart
+        self.redpanda.restart_nodes(self.redpanda.nodes)
+        self._check_value_everywhere("space_management_enable", True)
+        self._check_value_everywhere("retention_local_strict", True)
+
+    @cluster(num_nodes=3)
+    @parametrize(wipe_cache=True)
+    @parametrize(wipe_cache=False)
+    def test_removal_of_legacy_default_partial_defaulted(
+            self, wipe_cache: bool):
+        # in 23.1 space management feature does not exist
+        old_version, _ = self.installer.latest_for_line(self.legacy_version)
+        self.installer.install(self.redpanda.nodes, old_version)
+        self.redpanda.start()
+
+        # in 23.2 space management exists, but is disabled by default for
+        # upgraded clusters.
+        self._upgrade(wipe_cache, self.intermediate_version)
+        self._check_value_everywhere("space_management_enable", False)
+        self._check_value_everywhere("retention_local_strict", True)
+
```

Comment:
What about clusters that started on 23.2, where `retention_local_strict` is False because that's the default? We're intentionally flipping that off -- is that right?

Reply:
Are you asking whether `retention_local_string` stays False after a 23.2 -> 23.3 upgrade in this case?

Reply:
i think the issue here is two fold. first is that this keeps strict retention enabled for new installations, which isn't what we want. maybe it's just one fold

Reply:
Kind of, I'm pointing out that the default in 23.2 is False, and it isn't quite captured by the commit message that this is an intentional flip in behavior in upgrade.

I chatted with Noah a bit, and it seems like this blanket setting of `retention_local_strict=True` isn't what we want, and we should perhaps revert the removal of the legacy default decorator. Otherwise, it's a bit convoluted to describe the upgrade behavior of this config with this change as is.

```
new 23.3 cluster (strict=True): I think the desired behavior is actually strict=False
23.2 -> 23.3 (strict=False -> strict=True): I don't think this is what we want either. It should leave things as strict=False moving forward
23.1 -> 23.2 -> 23.3 (strict=True -> strict=True -> strict=True): this is okay
```